File: 0x10-python-network_1/103-search_twitter.py
#!/usr/bin/python3
"""
This module takes in 3 strings and sends a search request to the Twitter API
"""
import requests
import sys
import base64


if __name__ == "__main__":
    bearer_token = "%s:%s" % (sys.argv[1], sys.argv[2])
    token_64 = "Basic %s" % base64.b64encode(bearer_token.encode())
    header = {"Content-Type":
              "application/x-www-form-urlencoded;charset=UTF-8",
              "Authorization": token_64}
    data = {"grant_type": "client_credentials"}
    req = requests.post('https://api.twitter.com/oauth2/token', headers=header,
                        data=data)
    access = req.json().get("access_token")
    url = "https://api.twitter.com/1.1/search/tweets.json"
    headers = {"Authorization": "Bearer %s" % access}
    query = {'q': sys.argv[3], 'count': 5}
    search_req = requests.get(url, params=query, headers=headers)
    search_req = search_req.json().get('statuses')
    for each in search_req:
        print("[{}] {} by {}".format(
            each['id'], each['text'], each['user']['name']))

File: 0x12-javascript-web_scraping/103-search_twitter.js
#!/usr/bin/node

let request = require('request');
let base64 = require('base-64');
let utf8 = require('utf8');

let arg = process.argv[4];
arg = arg.slice(0, 0) + arg.slice(1, arg.length);

let url = 'https://api.twitter.com/1.1/search/tweets.json?q=%23' + arg;

let oauth = process.argv[2] + ':' + process.argv[3];
oauth = utf8.encode(oauth);
oauth = base64.encode(oauth);
oauth = utf8.decode(oauth);

let header = {
  'Authorization': 'Basic ' + oauth,
  'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
};

let data = {
  'grant_type': 'client_credentials'
};

let tokens = {
  url: 'https://api.twitter.com/oauth2/token',
  headers: header,
  form: data
};

request.post(tokens, function (error, response, body) {
  if (error) {
    console.log(error);
  }
  let data = JSON.parse(body);
  let access = data['access_token'];
  let searchHead = {
    'Authorization': 'Bearer ' + access
  };
  request.get({url: url, headers: searchHead}, function (error, response, body) {
    if (error) {
      console.log(error);
    }
    let hashTags = JSON.parse(body)['statuses'];
    for (let i = 0; i < 5; i++) {
      console.log('[' + hashTags[i]['id_str'] + '] ' + hashTags[i]['text'] +
        ' by ' + hashTags[i]['user']['name']);
    }
  });
});
